velocitas_lib/src/velocitas_lib/services.py

```diff
@@ -0,0 +1,144 @@
+    """Parse service spec configuration and return it as an named tuple.
+
+    Args:
+        service_spec_config: The specificon of the services from config file.
+    """
+
+    is_enabled = True
+    use_dapr = True
+    container_image = None
+    env_vars = dict[str, Optional[str]]()
+    ports = []
+    port_forwards = []
+    mounts = []
+    args = []
+    patterns = []
+
+    variables = ProjectVariables()
+
+    for config_entry in service_spec_config:
+        key = config_entry["key"]
+        value = config_entry["value"]
+
+        if isinstance(value, str):
+            value = variables.replace_occurrences(value)
+
+        if key == "enabled":
+            is_enabled = value is True or value == "true"
+        elif key == "image":
+            container_image = value
+        elif key == "env":
+            pair = value.split("=", 1)
+            inner_key = pair[0].strip()
+            env_vars[inner_key] = None
+            if len(pair) > 1:
+                env_vars[inner_key] = pair[1].strip()
+        elif key == "no-dapr":
+            if isinstance(value, str):
+                use_dapr = not value == "true"
+            elif isinstance(value, bool):
+                use_dapr = not value
+            else:
+                raise ValueError("Unsupported value type!")
+        elif key == "arg":
+            args.append(value)
+        elif key == "port":
+            ports.append(value)
+        elif key == "port-forward":
+            port_forwards.append(value)
+        elif key == "mount":
+            mounts.append(value)
+        elif key == "start-pattern":
+            patterns.append(value)
+
+    if container_image is None:
+        raise RuntimeError("Container image needs to be provided!")
+
+    return ServiceSpecConfig(
+        image=container_image,
+        is_enabled=is_enabled,
+        env_vars=env_vars,
+        use_dapr=use_dapr,
+        args=args,
+        ports=ports,
+        port_forwards=port_forwards,
+        mounts=mounts,
+        startup_log_patterns=patterns,
+    )
+
+
+def get_services() -> List[Service]:
+    """Return all specified services as Python object."""
+    path = Path(f"{get_package_path()}/runtime.json")
+    variable_value = require_env("runtimeFilePath")
+
+    if variable_value is not None:
+        overwritten_path = Path(variable_value)
+        if not overwritten_path.is_absolute():
+            overwritten_path = Path(get_workspace_dir()).joinpath(overwritten_path)
+
+        if overwritten_path.exists():
+            path = overwritten_path
+            print(f"runtime.json path redirected to {path}")
+
+    json_array: List[Dict] = json.load(
+        open(
+            path,
+            encoding="utf-8",
+        )
+    )
+
+    services: List[Service] = list()
+    for service_json in json_array:
+        service_id = service_json["id"]
+        # service_interfaces = service_json['interfaces']
+        service_config = ServiceSpecConfig(image="none")
```

Comment:
Why setting this image to string none?
```suggestion
        service_config = ServiceSpecConfig()
```

Reply:
Image is a mandatory parameter, if we set this to None, the typing would have to be `Optional[str]` but all of our current code relies on image to be present. But you are right, this should be handled differently.